kiwi_lint/generic_foreign_key.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+from pylint.checkers import utils, BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint_django.utils import node_is_subclass
+
+
+class GenericForeignKeyChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+
+    name = 'generic-foreign-key-checker'
+
+    msgs = {
+        'R4525': (
+            'Avoid using GenericForeignKey',
+            'avoid-generic-foreign-key',
+            'Avoid using django.contrib.contenttypes.fields.GenericForeignKey',
+        )
+    }
+
+    @utils.check_messages('avoid-generic-foreign-key')
+    def visit_classdef(self, node):
+
+        if not node_is_subclass(node, 'django.db.models.base.Model'):
```

Comment:
why not search for sub-classes of GenericForeignKey instead ? That will require different callbacks/visit_ methods though.

Reply:
Thank you for your feed back.
I told this will be more performant as it only checks Model definitions.
I will give that a go. Thank you.

Reply:
I need to know are you going to switch to a different implementation or no b/c the last comment indicates there would be something new but the PR doesn't show that. 

Let me know in which direction is this going ?